core/Decode.hpp

```diff
@@ -25,62 +34,307 @@ namespace olympia
+    //!
+    //! This is a run-time determined value extracted from parsing 
+    //! the JSON files. Future feature: FSL file support 
+    sparta::Counter fusion_num_groups_defined_;
+
+    //! \brief the number of fusion groups utilizaed
+    //!
+    //! Some number of run-time or compile-time fusion groups
+    //! are defined. 
+    //! 
+    //! This is a measure of how many defined groups were
+    //! used during execution. A fusion group used 100 times
+    //! counts only once.
+    sparta::Counter fusion_num_groups_utilized;
+
+    //! \brief the predicted number of cycles saved by fusion
+    //!
+    //! For the simplist case a fusion group of N instructions
+    //! saves N-1 cycles. Modified for the cases where some
+    //! instructions and some fusion groups are multi-cycle.
+    //!
+    //! This is a prediction without concern for pipeline
+    //! behavior. This is compared against measured cycles
+    //! with fusion enabled vs. disabled to show effectiveness
+    //! of the defined fusion groups.
+    sparta::Counter fusion_pred_cycles_saved_;
+
+    //! \brief temporary for number of instructions to decode this time
+    const uint32_t num_to_decode_;
+
+    //! \brief master fusion enable data member
+    //! 
+    //! This is the model's fusion enable parameter. When not enabled
+    //! the fusion methods have no impact on function or performance
+    //! of the model. \see fusion_enable_register_;
+    const bool     fusion_enable_{false};
+
+    //! \brief represents the FER in h/w
+    //!
+    //! The register holds an encoded value. Each fusion group 
+    //! has an associated page, and an enable. The upper 4 bits
+    //! represent a page, the lower 28 bits are the enables. 
+    //! PAGE=0xF all enabled, PAGE=0x0 all disabled. This is preferred
+    //! over a separate page register and enable register.
+    //!
+    //! Currently defined but not used
+    const uint32_t fusion_enable_register_{0};
+
+    //! \brief maximum allowed latency induced by fusion
+    //!
+    //! This typically zero, a parameter is provided to allow
+    //! experimenting with instruction gathering
+    const uint32_t fusion_max_latency_{0};
+
+    //! \brief max attempts to match UIDs of instrs in the XIQ
+    const uint32_t fusion_match_max_tries_{0};
+
+    //! \brief the longest chain of UIDs across all groups
+    //!
+    //! \see FusionGroup
+    uint32_t fusion_max_group_size_{0};
+
+    //! \brief running count of fusion latency
+    uint32_t latency_count{0};
+
+    //! \brief the Fusion API instance
+    //!
+    //! \see Fusion
+    FusionType *fuser{nullptr};
+    fusion::HCache hcache;
+
+    //! \brief fusion function object callback proxies
+    struct cbProxy;
+   //! \brief the set of implemented fusable sequences 
+   static FusionType::FusionGroupCfgListType fusionGroupCfgs;
+   //! \brief this counts the number of times a group is used
+    std::map<std::string,uint32_t> matchedFusionGroups;
```

Comment:
Nit: naming doesn't match the rest of the code.

Reply:
fixed